bittide-instances/src/Bittide/Instances/Hitl/Transceivers.hs

```diff
@@ -84,14 +112,43 @@ goTransceiversUpTest refClk sysClk rst rxns rxps miso =
   -- Transceiver setup
   gthAllReset = unsafeFromActiveLow spiDone
 
+  -- Send counters
+  counters =
+    zipWith3
+      counter
+      transceivers.txClocks
+      transceivers.txResets
+      transceivers.txSamplings
+
+  expectCounterError =
+    zipWith3
+      expectCounter
+      transceivers.rxClocks
+      transceivers.rxResets
+      transceivers.rxDatas
+
+  expectCounterErrorSys =
+      fmap and
+    $ bundle
+    $ zipWith (.&&.) transceivers.linkUps
+    $ zipWith (`xpmCdcSingle` sysClk) transceivers.rxClocks expectCounterError
```

Comment:
Not only should we check whether every digit is as expected, we also need to verify that we see user data in the first place. I.e., make the test fail if we didn't count to at least N.